### PR TITLE
GEODE-8721: member that should become coordinator never detects loss of current coordinator

### PR DESCRIPTION
**Pull request by bschuchardt**
If a server is in the process of performing an availability check on
another server we shouldn't update the contact timestamp for
the suspected server based on gossip from another server.  Doing so
will make the availability check pass and send out another gossip
message that would likewise update their timestamps for the suspected
server, perpetuating the notion that the suspect is still around.

Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

@kamilla1201 

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [x] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [x] Is your initial contribution a single, squashed commit?

- [x] Does `gradlew build` run cleanly?

- [x] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, check Concourse for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

File: geode-membership/src/main/java/org/apache/geode/distributed/internal/membership/gms/fd/GMSHealthMonitor.java [hunk 1]
```diff
@@ -152,7 +152,7 @@
   /**
    * Members undergoing final checks
    */
-  private final List<ID> membersInFinalCheck =
+  final List<ID> membersInFinalCheck =
```
Comment (kohlmu-pivotal) [inline geode-membership/src/main/java/org/apache/geode/distributed/internal/membership/gms/fd/GMSHealthMonitor.java:155]: do we need to add `@VisibleForTesting` annotation here?

File: geode-membership/src/main/java/org/apache/geode/distributed/internal/membership/gms/fd/GMSHealthMonitor.java [hunk 2]
```diff
@@ -130,7 +130,7 @@
   public static final long MEMBER_SUSPECT_COLLECTION_INTERVAL =
       Long.getLong("geode.suspect-member-collection-interval", 200);
 
-  private volatile long currentTimeStamp;
+  volatile long currentTimeStamp;
```
Comment (kohlmu-pivotal) [inline geode-membership/src/main/java/org/apache/geode/distributed/internal/membership/gms/fd/GMSHealthMonitor.java:133]: Should we maybe remove direct access to this field and rather replace it with an accessor that we can then mark with `@VisibleForTesting`?

File: geode-membership/src/main/java/org/apache/geode/distributed/internal/membership/gms/fd/GMSHealthMonitor.java (hunk 1 above)
Comment (bschuchardt) [reply geode-membership/src/main/java/org/apache/geode/distributed/internal/membership/gms/fd/GMSHealthMonitor.java:155]: yes

File: geode-membership/src/main/java/org/apache/geode/distributed/internal/membership/gms/fd/GMSHealthMonitor.java (hunk 2 above)
Comment (bschuchardt) [reply geode-membership/src/main/java/org/apache/geode/distributed/internal/membership/gms/fd/GMSHealthMonitor.java:133]: seems the same as membersInFinalCheck to me, allowing a test subclass to access it

File: geode-membership/src/main/java/org/apache/geode/distributed/internal/membership/gms/fd/GMSHealthMonitor.java [hunk 3]
```diff
@@ -1242,7 +1243,11 @@ void processMessage(FinalCheckPassedMessage<ID> m) {
     if (isStopping) {
       return;
     }
-    contactedBy(m.getSuspect());
+    // if we're currently processing a final-check for this member don't artificially update the
```
Comment (echobravopapa) [inline geode-membership/src/main/java/org/apache/geode/distributed/internal/membership/gms/fd/GMSHealthMonitor.java:1252]: solid fix!

**Review by echobravopapa — APPROVED** [review]

**Pull request merged**